localstack/services/s3/s3_listener.py

```diff
@@ -412,10 +413,17 @@ def return_response(self, method, path, data, headers, response):
 
         bucket_name = get_bucket_name(path, headers)
 
+        # No path-name based bucket name?  Try host-based
+        hostname_parts = headers['host'].split('.')
+        if (not bucket_name or len(bucket_name) == 0) and len(hostname_parts) > 1:
```

Comment:
nitpick: Could be simplified to:
```
if not bucket_name and len(hostname_parts) > 1:
```
(because `len(bucket_name) == 0` implies `not bucket_name`)